server/activate_hooks.go

```diff
@@ -3,41 +3,59 @@ package main
 import (
 	"fmt"
 
+	"github.com/blang/semver"
 	"github.com/mattermost/mattermost-server/model"
 )
 
+const minimumServerVersion = "5.4.0"
+
 // OnActivate is invoked when the plugin is activated.
 //
 // This demo implementation logs a message to the demo channel whenever the plugin is activated.
 func (p *Plugin) OnActivate() error {
-	teams, err := p.API.GetTeams()
+	v, err := semver.Parse(p.API.GetServerVersion())
 	if err != nil {
 		p.API.LogError(
-			"failed to query teams OnActivate",
+			"failed to parse server version",
 			"error", err.Error(),
 		)
+		return err
+	}
+	r := semver.MustParseRange(">=" + minimumServerVersion)
+	if !r(v) {
+		return fmt.Errorf("current Mattermost version is to low. Please update your Mattermost Server to a least v%s.", minimumServerVersion)
```

Comment:
In this particular case, we'll actually fail on line 17 above given that `GetServerVersion` won't be implemented prior to `5.4.0`. This seems like a good opportunity to teach a potential user of the API about this nuance. 

What if we change the behaviour here not to enforce a minimum, but to modify the `OnActivate` message posted to the channel below to dump the `model.Version` against which the plugin was compiled alongside the result from  `p.API.GetServerVersion()`, displaying `unknown` if the API isn't actually available as such (because an error was returned). We use this plugin during release testing, and I think it would be more useful to keep the plugin "working" across versions (with some useful version dumping to prove the API call is correct) than failing outright.

It might also be interesting to formalize the error returned when an API isn't implemented, but that's an idea for another PR I suppose :)

Reply:
Unfortunately, it doesn't panic: just an error log at present, returning zero values if any. I'm assuming the semver would fail on attempting to parse the resulting empty string.

I do agree that there's a dual audience here, and that we should favour plugin developers. Let's stick with your proposal.